Zend/zend_execute.c

```diff
@@ -1932,12 +1932,12 @@ static zend_always_inline void zend_fetch_property_address(zval *result, zval *c
 			}
 
 			/* this should modify object only if it's empty */
-			if (type != BP_VAR_UNSET &&
-			    EXPECTED(Z_TYPE_P(container) <= IS_FALSE ||
-			      (Z_TYPE_P(container) == IS_STRING && Z_STRLEN_P(container)==0))) {
-				zval_ptr_dtor_nogc(container);
-				object_init(container);
+			if (type != BP_VAR_UNSET) {
+				if (!make_real_object(container)) {
```

Comment:
I presume the logic in this function was duplicated here because the first branch in it should always be taken in this case. If that's true, then perhaps just adding a `zend_error` to the original code would be best:
```diff
				zval_ptr_dtor_nogc(container);
				object_init(container);
+				zend_error(E_WARNING, "Creating default object from empty value");
```

If that's not the case, then the logic would look cleaner without a goto jump:
```
if (type == BP_VAR_UNSET || !make_real_object(container)) {
```

Reply:
We could just as easily add the error here as well, but I went with the reuse the existing function route.  I assume this object_init() branch here existed before the function existed, probably explains the bug report.  But yes, i mind-farted there, and will apply that fix.